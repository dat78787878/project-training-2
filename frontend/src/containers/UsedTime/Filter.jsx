import React from 'react';
import { Col, Form, Row } from 'react-bootstrap';
import { v4 as uuidv4 } from 'uuid';

const Filter = ({
  setDataRender,
  usedTimeData,
  search,
  date,
  type,
  setSearch,
  setDate,
  setType
}) => {
  const listOS = ['Android', 'iOS'];

  const handleSearch = (e) => {
    e.preventDefault();
    const keyFilter = e.target.value.trim().toLowerCase();
    setSearch(keyFilter);
  };

  const handleSelect = (e) => {
    setDataRender(usedTimeData);
    setType(e.target.value);
  };

  const handleChange = (e) => {
    e.preventDefault();
    setDate(e.target.value);
  };

  return (
    <div className="filter">
      <Row>
        <Col>
          <input
            data-testid="input-name-used"
            className="filter-input"
            onChange={handleSearch}
            placeholder="Search"
            value={search}
          />
        </Col>
        <Col>
          <div className="d-flex input-date">
            <input
              data-testid="input-date-used"
              type="date"
              name="trip-start"
              value={date}
              onChange={handleChange}
            />
          </div>
        </Col>
        <Col>
          <Form.Select
            aria-label="Default select example"
            data-testid="select-box-used"
            onChange={handleSelect}
            value={type}>
            <option>Please select</option>
            {listOS.map((val, index) => {
              return (
                <option key={index + uuidv4()} value={val}>
                  {val}
                </option>
              );
            })}
          </Form.Select>
        </Col>
      </Row>
    </div>
  );
};

export default Filter;
