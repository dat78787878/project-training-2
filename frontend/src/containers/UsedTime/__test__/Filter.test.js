import React from 'react';
import { render } from '@testing-library/react';
import Filter from '../Filter';
import { BrowserRouter } from 'react-router-dom';
window.React = React;

describe('Loading component', () => {
  it('render component success', () => {
    const { container } = render(
      <BrowserRouter>
        <Filter />
      </BrowserRouter>
    );
    expect(container.getElementsByClassName('filter'));
    expect(container).toMatchSnapshot();
  });
});
