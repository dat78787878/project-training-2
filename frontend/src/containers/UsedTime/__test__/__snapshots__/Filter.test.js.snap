// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component render component success 1`] = `
<div>
  <div
    class="filter"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <input
          class="filter-input"
          data-testid="input-name-used"
          placeholder="Search"
          value=""
        />
      </div>
      <div
        class="col"
      >
        <div
          class="d-flex input-date"
        >
          <input
            data-testid="input-date-used"
            name="trip-start"
            type="date"
            value=""
          />
        </div>
      </div>
      <div
        class="col"
      >
        <select
          aria-label="Default select example"
          class="form-select"
          data-testid="select-box-used"
        >
          <option>
            Please select
          </option>
          <option
            value="Android"
          >
            Android
          </option>
          <option
            value="iOS"
          >
            iOS
          </option>
        </select>
      </div>
      <div
        class="col"
      >
        <button
          aria-label="button-name"
          class="btn btn-success"
          data-testid="add-button"
          type="button"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;
